lib/graphql/execution/interpreter/runtime.rb

```diff
@@ -8,6 +8,17 @@ class Interpreter
       #
       # @api private
       class Runtime
+
+        class GraphQLResultHash < Hash
+          def graphql_metadata
```

Comment:
I was thinking separate accessor methods rather than putting an extra hash in your hash :P

Reply:
:+1: Agreed, that'd be less error prone and just as fast.

Reply:
Done in https://github.com/rmosolgo/graphql-ruby/pull/3494/commits/96851cf73fa7ccb0dd4021d0d3f90fcec81b0d25 ... and now there's a subclass of Array, too 🙈 !